### PR TITLE
Added delta and fixed some problems with y-file generator

### PR DESCRIPTION
**Pull request by amunozj**
- Now the script is properly tied to the final image of each sequence.
- I fixed a problem with the 'get_Y_index' index finder that was returning the wrong flux.
- Added some more comments and documentation.
- Now the script also generates a column with the delta.
- Renamed one of the variables (duration to catch) to better match our terminology.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -105,4 +105,8 @@ ENV/
 
 # mypy
 .mypy_cache/
+<<<<<<< HEAD
```
Comment (smcgregor) [inline .gitignore]: It looks like you have an unresolved diff here. You should remove the lines with `<<<`, `====`, and `>>>>`.

File: tools/xray_flux_forecast/v4_find_ys.py [hunk 2]
```diff
@@ -33,22 +40,30 @@
...
+#                             #
+#  Delay and Catch definition #
+#                             #
+###############################
+
 delays = [60]
-durations = [12, 36, 60, 24*60]
-#durations = [12, 36, 60, 24*60]
+catches = [12, 36, 60, 24*60]
+#catches = [12, 36, 60, 24*60]
```
Comment (smcgregor) [inline tools/xray_flux_forecast/v4_find_ys.py]: Duplicate line?

File: tools/xray_flux_forecast/v4_find_ys.py [hunk 3]
```diff
@@ -33,22 +40,30 @@
...
 
 for delay in delays:
-    for duration in durations:
+    for catch in catches:
         
         def get_Y_index(date):
+            """Function that finds the index on the large y file for a given date"""
```
Comment (smcgregor) [inline tools/xray_flux_forecast/v4_find_ys.py]: Good function comments have a brief description, followed by the type definition and description of the parameters. I cannot read this comment and understand the purpose of the function, what code will likely call the function, and how I should call the function if I want to adapt it to another use in the future.

File: tools/xray_flux_forecast/v4_find_ys.py [hunk 4]
```diff
@@ -58,14 +73,17 @@ def get_yval(filename, index):
                     this_arr.append(0.0)
             return max(this_arr)
         
+        #Adding column names
         Y_vals = [] #[[Date, Y-data - 12min/36min/1hr/24hr max, Channel 0,7 coefficients]]
         this_term = ['Date Stamp']
```
Comment (smcgregor) [inline tools/xray_flux_forecast/v4_find_ys.py]: I would label the local accumulator something more descriptive, like `y_row`. In this case you would likely be better off doing this whole operation in one line, i.e. this_term `['Date Stamp', 'Flux', 'Delta', 'Filename']. Alternatively, you could append to Y_vals directly and avoid the assignment to `this_term`.

**Review by smcgregor — COMMENTED** [review]
I left a few comments below after a brief review for style. I will merge the PR tomorrow morning regardless of whether you address the issues I highlight. The only thing that I currently see that must change is the .gitignore.

**Comment by amunozj** [comment]
thank you Sean!  I will spend 20 minutes making it better and learning from your suggestions.  I understand it's not high priority, but I want to learn how to do it better

**Pull request merged**